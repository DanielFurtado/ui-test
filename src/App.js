import { Routes, Route } from 'react-router-dom';
import { QueryClient, QueryClientProvider } from 'react-query';

import Gallery from './routes/gallery/gallery.component';
import Navigation from './routes/navigation/navigation.component';
import SignUp from './routes/sign-up/sign-up.component';

import './App.scss';

const queryClient = new QueryClient();

const App = () => {
  return (
    <QueryClientProvider client={queryClient}>
      <Routes>
        <Route path='/' element={<Navigation />}>
          <Route index element={<Gallery />} />
          <Route path='sign-up' element={<SignUp />} />
        </Route>
      </Routes>
    </QueryClientProvider>
  );
}

export default App;
