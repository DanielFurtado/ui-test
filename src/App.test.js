import {BrowserRouter as Router} from 'react-router-dom';
import renderer from 'react-test-renderer';
import App from './App';

it('renders', () => {
  const component = renderer.create(
    <Router>
      <App />
    </Router>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
