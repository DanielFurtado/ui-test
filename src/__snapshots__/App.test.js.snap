// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
Array [
  <header
    className="navigation"
  >
    <nav
      className="nav-container"
    >
      <a
        aria-current="page"
        className="logo-container active"
        href="/"
        onClick={[Function]}
      >
        <svg
          className="logo"
        >
          motorway-logo.svg
        </svg>
      </a>
      <ul
        className="nav-links-container"
      >
        <li>
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Gallery
          </a>
        </li>
        <li>
          <a
            className="nav-link"
            href="/sign-up"
            onClick={[Function]}
          >
            Sign up
          </a>
        </li>
      </ul>
    </nav>
  </header>,
  <h3
    className="loading"
  >
    Loading
    <div
      className="lds-ellipsis"
    >
      <div />
      <div />
      <div />
      <div />
    </div>
  </h3>,
]
`;
